js/src/components/gtin-migration-banner/gtin-migration-banner.js

```diff
@@ -0,0 +1,181 @@
+		return null;
+	}
+
+	const closeModal = () => {
+		recordGlaEvent( 'gla_modal_closed', {
+			context: GTIN_MIGRATION_BANNER_CONTEXT,
+		} );
+		setShowModal( false );
+	};
+
+	const openModal = () => {
+		recordGlaEvent( 'gla_modal_open', {
+			context: GTIN_MIGRATION_BANNER_CONTEXT,
+		} );
+		setShowModal( true );
+	};
+
+	const handleStartMigrationClick = async () => {
+		recordGlaEvent( 'gla_gtin_migration_banner_migration_start', {
+			context: GTIN_MIGRATION_BANNER_CONTEXT,
+		} );
+		reset();
+		try {
+			await startMigration( { parse: false } );
+			recordGlaEvent( 'gla_gtin_migration_banner_migration_scheduled', {
+				context: GTIN_MIGRATION_BANNER_CONTEXT,
+			} );
+			setShowModal( false );
+			setStatus( GTIN_MIGRATION_STARTED );
+			createNotice(
+				'info',
+				__(
+					'GTIN Migration was successfully scheduled.',
+					'google-listings-and-ads'
+				)
+			);
+		} catch ( e ) {
+			recordGlaEvent( 'gla_gtin_migration_banner_migration_failed', {
+				context: GTIN_MIGRATION_BANNER_CONTEXT,
+				error: e.message,
+			} );
+			createNotice(
+				'error',
+				__(
+					'Unable to start GTIN Migration.',
+					'google-listings-and-ads'
+				)
+			);
+		}
+	};
+
+	return (
+		<>
+			{ showModal && (
+				<AppModal
+					className="gla-gtin-migration-banner-modal"
+					title={ __(
+						'Before you start the migration…',
+						'google-listings-and-ads'
+					) }
+					buttons={ [
+						<AppButton key="1" isSecondary onClick={ closeModal }>
+							{ __( 'Never mind', 'google-listings-and-ads' ) }
+						</AppButton>,
+						<AppButton
+							key="2"
+							disabled={ loading }
+							isPrimary
+							onClick={ handleStartMigrationClick }
+						>
+							{ __(
+								'Start migration',
+								'google-listings-and-ads'
+							) }
+						</AppButton>,
+					] }
+					onRequestClose={ closeModal }
+				>
+					<p>
+						{ createInterpolateElement(
+							__(
+								"This migration will copy all GTIN numbers set in the Google for WooCommerce Product tab into the new GTIN field under the Product Inventory tab. If you have already set GTIN numbers in some of your products' Inventory tab, they will not be overridden. The GTIN numbers in the Google for WooCommerce tab will not be removed. The migration will run in the background and is not reversible. You can check the migration process on the <link>WooCommerce Scheduled Actions page</link>.",
+								'google-listings-and-ads'
+							),
+							{
+								link: <LinkGTINMigrationJobStatusPage />,
+							}
+						) }
+					</p>
+				</AppModal>
+			) }
+			{ status === GTIN_MIGRATION_READY && (
+				<Notice isDismissible={ false }>
+					{ createInterpolateElement(
+						__(
+							"The GTIN field managed by WooCommerce in the Product's inventory section, will now be used by Google for WooCommerce. It will continue to support the previous field and any mapping rules you have setup for the GTIN field. If you would like to migrate the data <link>click here.</link>",
```

Comment:
I realize we aren't 100% consistent on this, but isn't it recommended to keep the punctuation outside of the link?

```suggestion
							"The GTIN field managed by WooCommerce in the Product's inventory section, will now be used by Google for WooCommerce. It will continue to support the previous field and any mapping rules you have setup for the GTIN field. If you would like to migrate the data <link>click here</link>.",
```